lib/ansible/modules/notification/mattermost.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Benjamin Jolivot <bjolivot@gmail.com>
+# Inspired by slack module :
+#    # (c) 2017, Steve Pletcher <steve@steve-pletcher.com>
+#    # (c) 2016, René Moser <mail@renemoser.net>
+#    # (c) 2015, Stefan Berggren <nsg@nsg.cc>
+#    # (c) 2014, Ramon de la Fuente <ramon@delafuente.nl>)
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = """
+module: mattermost
+short_description: Send Mattermost notifications
+description:
+    - The M(mattermost) module sends notifications to U(http://your.mattermost.url) via the Incoming WebHook integration
```

Comment:
IMO it is probably better to use C(mattermost) instead, because the whole point of using M(something) is to have a link to the module's documentation. But that's where people already are when seeing the link. So a link pointing to itself is pretty moot.